scripts/Packages/Package_Gullwing__QFP_SOIC_SO/Readme.md

```diff
@@ -0,0 +1,100 @@
+# IPC gullwing generator
+
+This generator uses IPC-7351B equations and by default fillet definitions to derive a footprint for gullwing stile packages.
+Examples of supported packages include: QFN, SOIC, SO
+
+## Running the script
+
+The script requires python 3.4 or newer. Run it with:
+`python3 ipc_gullwing_generator.py size_definitions/eqfp.yaml` (replace eqfp.yaml with the size definition file that contains your part.)
+
+## Size definition format
+
+Every file contains a header with parameters applied to all parts. These define the common footprint name prefix and the output library.
+
+```yaml
+FileHeader:
+  library_Suffix: 'QFP' #resulting library name Package_QFP
+  device_type: 'EQFP' #footprint names will be of style ...EQFP-pincount_...
+```
+
+---
+
+Every further entry in the script is assumed to be a package size definition. The top level parameter will give the internal parameter set name. This one must be unique in this file and should be representative of the footprint as it will be used in error messages. (It also makes later maintenance easier if it is easy to determine which footprint was genererated by which parameter set.)
+
+``` yaml
+internal_name:
+  # parameter list
+```
+
+The following parameters are supported for every package
+- Size source to be added to footprint documentation field (`size_source`) {url}
```

Comment:
'url' here doesn't correspond to a data type, but 'string' and'int' and 'float' are used below and does.  'dimension' is used below and doesn't. Perhaps this needs to include data type and also a description, like 'string, url' or 'int/float, dimension'?

Reply:
url can be seen as an abstract datatype. (It is expected to be a string representing an url. Just because i do not check this for validity does not mean it is not expected to be an url)

Reply:
Because you're mixing the content of the field and the field type it's messy.

The URL is a string with a specific format and so the `size_source` field takes a string type.

In addition, generally it will be package documentation and not footprint documentation (special footprint info may be included).

Also, adding a colon as you said to separate the field and info about it would be nice.

Maybe 'Size source: link to datasheet and optionally other package documentation (`size_source`) {string}'? Things could be reformatted to avoid duplication or near-duplication of info if the following format is adopted: '`size_source`: link to datasheet and optionally other package documentation {string}'.

Reply:
I document what the intention of the file format is. What it can be misused for is something that the user will need to check for themselfes (again the documentation is about the generator not how it is used for the official library.)